virttest/asset.py

```diff
@@ -306,20 +307,27 @@ def download_test_provider(provider, update=False):
         ref = provider_info.get('ref')
         pubkey = provider_info.get('pubkey')
         download_dst = data_dir.get_test_provider_dir(provider)
-        repo_downloaded = os.path.isdir(os.path.join(download_dst, '.git'))
-        original_dir = os.getcwd()
-        if not repo_downloaded or update:
-            download_dst = git.get_repo(uri=uri, branch=branch, commit=ref,
-                                        destination_dir=download_dst)
+        try:
+            repo_downloaded = os.path.isdir(os.path.join(download_dst, '.git'))
+            original_dir = os.getcwd()
+            if not repo_downloaded or update:
+                download_dst = git.get_repo(uri=uri, branch=branch, commit=ref,
+                                            destination_dir=download_dst)
+                os.chdir(download_dst)
+                try:
+                    process.run('git remote add origin %s' % uri)
+                except process.CmdError:
+                    pass
+                process.run('git pull origin %s' % branch)
             os.chdir(download_dst)
-            try:
-                process.run('git remote add origin %s' % uri)
-            except process.CmdError:
-                pass
-            process.run('git pull origin %s' % branch)
-        os.chdir(download_dst)
-        process.system('git log -1')
-        os.chdir(original_dir)
+            process.system('git log -1')
+            os.chdir(original_dir)
+        except (KeyboardInterrupt, process.CmdError):
+            logging.error('Cleaning up provider %s download dir %s', provider,
+                          download_dst)
+            if os.path.isdir(download_dst):
+                shutil.rmtree(download_dst)
```

Comment:
I have important data in my test provider download dir, because that's where I develop new test code. I would be very upset if you deleted it because something went wrong while running `git log -1`.

IMO the most important thing to do here is to explain which git repository looks corrupted (because `git log -1` failed), and let the user delete it if they think it is OK. Later we may have very specific code that tries to recover from an interrupted git-clone, but I would never try to recover by deleting data if the user hasn't asked for it explicitly.